src/core/control/tools/SplineHandler.cpp

```diff
@@ -204,87 +149,102 @@ auto SplineHandler::onMotionNotifyEvent(const PositionInputData& pos, double zoo
         this->buttonDownPoint = Point(pos.x / zoom, pos.y / zoom);
         this->currPoint = snappingHandler.snap(this->buttonDownPoint, knots.back(), pos.isAltDown());
     }
+    rg = rg.unite(this->computeLastSegmentRepaintRange());
 
-    rect.unite(this->computeRepaintRectangle());
-    this->redrawable->repaintRect(rect.x, rect.y, rect.width, rect.height);
-
+    this->viewPool->dispatch(xoj::view::SplineToolView::FLAG_DIRTY_REGION, rg);
     return true;
 }
 
-void SplineHandler::onSequenceCancelEvent() { stroke.reset(); }
+void SplineHandler::onSequenceCancelEvent() {
+    //  Touch screen sequence changed from normal to swipe/zoom/scroll sequence
```

Comment:
This comment is misleading as `onSequenceCancelEvent` can also happen without touchscreen.